aws-resources/src/main/java/io/opentelemetry/contrib/aws/resource/CloudResourceProvider.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.aws.resource;
+
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import io.opentelemetry.sdk.autoconfigure.spi.internal.ConditionalResourceProvider;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.semconv.ResourceAttributes;
+
+public abstract class CloudResourceProvider implements ConditionalResourceProvider {
```

Comment:
My preference would be to make `CloudResourceProvider` and interface and have the `shouldApply` be a default method. I just get twitchy around class inheritance. Burned by it too many times....but this is really small enough to hopefully not be a problem....but code changes over time. :)